### PR TITLE
added "Use in Neovim" to readme

### PR DESCRIPTION
**Pull request by Chris44442**
[Issue#266](https://github.com/VHDL-LS/rust_hdl/issues/266#issue-2164238029)

### CODE REVIEW

**Comment by Schottkyc137** [comment]
Thanks for the contribution!

At the moment, the instructions sound just a little bit too generic to me. Can you change the tone so that it is clear that this is only one (of possibly many) options to enable the language server? For example, there is a [Pull Request](https://github.com/mason-org/mason-registry/pull/4596) trying to add `vhdl_ls` to Mason.
Also, instead of using `cargo`, one could compile from source or download the pre-built binaries.

File: README.md [hunk 1]
```diff
@@ -52,6 +52,29 @@ following snippet to your config:
 (vhdl-ext-lsp-set-server 've-rust-hdl)   ; `lsp' config
 ```
 
+## Use in Neovim
+
+Neovim provides an LSP client to the VHDL_LS language server. Install the VHDL_LS  
+binary e.g. via Cargo. The binary must be on the path and executable (if you  
+can run "vhdl_ls -h" in the terminal then you're good). Put the "vhdl_libraries"
```
Comment (Schottkyc137) [inline README.md:59]: I'm not quite sure everyone understands what The "vhdl_libraries" are. Could you add a small explanation here?

File: README.md [hunk 2]
```diff
@@ -52,6 +52,29 @@ following snippet to your config:
...
+function STARTVHDLLS()
+  vim.lsp.start({
+    name = 'vhdl_ls',
+    cmd = {'vhdl_ls'},
+  })
+end
+vim.api.nvim_set_keymap('n', '<F5>', ':lua STARTVHDLLS()<CR>', { noremap = true, silent = true })
+```
+
+This is just one example, there are many ways to run the language server. Anyway
```
Comment (Schottkyc137) [inline README.md:74]: What are the other ways to run the language server?

**Review by Schottkyc137 — CHANGES_REQUESTED** [review]
To elaborate more on my comment: I think this is more or less one example of how you could integrate `vhdl_ls` into neovim, correct? If so I'd appreciate it if the text would reflect this.

**Comment by Chris44442** [comment]
Oh, I didn't know about the [Mason PR](https://github.com/mason-org/mason-registry/pull/4596). Yeah, this would be the preferred way to do it of course. But until its merged, I guess it makes sense to describe a way, how to manually add the LS to Neovim. I know it would've saved me some time a few months back when I started using it and didn't know what I was doing.

> I think this is more or less one example of how you could integrate vhdl_ls into neovim, correct?

At the end of the day they all use the vim.lsp framework. But I will try to make it clear that there is an automatic way upcoming (mason), and that this here is the way to do it manually. If that is ok with you.

File: README.md (hunk 2 above)
Comment (Chris44442) [reply README.md:74]: What I meant is that you can also autorun the LS, or start it when you enter a buffer, or tie it to another plugin somehow. Essentially any way you like within what Neovim allows. But wanted to keep the "How to" short. Should I just remove this remark?

File: README.md (hunk 1 above)
Comment (Chris44442) [reply README.md:59]: Thanks for bringing that up, it is something I didn't quite fully understand. There is the [vhdl_libraries](https://github.com/VHDL-LS/rust_hdl/tree/master/vhdl_libraries)  directory in this repo. When I use VHDL_LS the way I described it expects this directory in the parent directory of the VHDL_LS binary, otherwise it will crash in Neovim. Simply copying it there worked for me, but I guess it is not exactly how it is supposed to be, right?

**Comment by Schottkyc137** [comment]
> At the end of the day they all use the vim.lsp framework. But I will try to make it clear that there is an automatic way upcoming (mason), and that this here is the way to do it manually. If that is ok with you.

Yea, that sounds good. My only concern is that the, way you described it, it sounds a bit like that's the only way of installing the client for nvim. I'm already happy with changing the title to `Manual Installation for NeoVim` (or something in that direction) to signify that there are other methods.

File: README.md (hunk 2 above)
Comment (Schottkyc137) [reply README.md:74]: I like it, but maybe rephrase it to something like "Using the example above, pressing F5 while inside NeoVim starts the language server. There are also other options, like automatically starting it when opening a certain file type, see the doc for more..."

File: README.md (hunk 1 above)
Comment (Schottkyc137) [reply README.md:59]: Well, yes and no :D
The language server needs the standard libraries to correctly work. That way it can, for example, resolve `std_logic` or other symbols from the standard library when you use them. So therefore simply copying is fine. However, the better approach would be to package the libraries with the executable. This is done for the artefacts and when you download a release, you get the packaged libraries with that release. However, using `cargo`, there is no standard way of adding additional files to the binary release and package them. Therefore, this is a bit awkward at the moment.

Comment (Chris44442) [reply README.md:59]: ohhh I see. Thanks for clearing that up. Maybe this should be noted somewhere else in the readme then? Because I am probably not the only one who installed it via cargo.

**Comment by Schottkyc137** [comment]
I'm happy with the changes now. This is a great addition for anyone trying to use Neovim with `vhdl_ls`. If you're happy too I will merge this.

File: README.md (hunk 1 above)
Comment (Schottkyc137) [reply README.md]: I agree. This should probably be noted down somewhere (preferably the readme) and the error message should point to the readme when the libraries cannot be found,

**Comment by Chris44442** [comment]
> If you're happy too I will merge this.

Yes, all good. Thanks!
